src/components/KMultiselect/KMultiselectItems.vue

```diff
@@ -0,0 +1,74 @@
+<template>
+  <KMultiselectItem
+    v-for="item, idx in nonGroupedItems"
+    :key="`${item.key ? item.key : idx}-item`"
+    :item="item"
+    @selected="handleItemSelect"
+  >
+    <template #content>
+      <slot
+        :item="item"
+        name="content"
+      />
+    </template>
+  </KMultiselectItem>
+  <div
+    v-for="grouping in groupings"
+    :key="`${grouping}-grouping`"
+    class="mb-1"
+  >
+    <span class="d-block k-multiselect-grouping-title">
+      {{ grouping }}
+    </span>
+    <KMultiselectItem
+      v-for="(item, idx) in getGroupingItems(grouping)"
+      :key="`${item.key ? item.key : idx}-item`"
+      :item="item"
+      @selected="handleItemSelect"
+    >
+      <template #content>
+        <slot
+          :item="item"
+          name="content"
+        />
+      </template>
+    </KMultiselectItem>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, PropType } from 'vue'
+import { MultiselectItem } from './KMultiselect.vue'
+import KMultiselectItem from '@/components/KMultiselect/KMultiselectItem.vue'
+
+interface MultiselectItemWithGrouping extends MultiselectItem {
+  grouping: string
+}
+
+const props = defineProps({
+  items: {
+    type: Array as PropType<MultiselectItem[]>,
+    default: () => [],
+    // Items must have a label & value
+    validator: (items: MultiselectItem[]) => !items.length || (items.every(i => i.label !== undefined && i.value !== undefined)),
+  },
+})
+
+const emit = defineEmits(['selected'])
+
+const handleItemSelect = (item: MultiselectItem, isNew?: boolean) => emit('selected', item, isNew)
+
+const nonGroupedItems = computed((): MultiselectItem[] => props.items.filter(item => !item.grouping))
+const groupings = computed((): string[] => [...new Set((props.items.filter(item => item.grouping) as unknown as MultiselectItemWithGrouping[]).map(item => item.grouping))].sort((a, b) => a.localeCompare(b.toLowerCase())))
```

Comment:
Yeah it is for ensuring uniqueness. What do you mean it doesn't guarantee uniqueness? [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set):
> The Set object lets you store unique values of any type, whether primitive values or object references.

Reply:
```ts
const mySet1 = new Set()
const newObj = { a: 1, b: 2 }
mySet1.add(newObj)

// newObj is referencing a different object, so this is okay
// This means you have two objects that are technically equal now in the Set
mySet1.add({ a: 1, b: 2 })
```

A good reference: https://medium.com/coding-at-dawn/how-to-use-set-to-filter-unique-items-in-javascript-es6-196c55ce924b#3e99

> When working with primitive values like [numbers](https://medium.com/javascript-in-plain-english/how-to-check-for-a-number-in-javascript-8d9024708153) or [strings](https://medium.com/javascript-in-plain-english/how-to-check-for-a-string-in-javascript-a16b196915ff), a Set object works just as you would expect it does — only unique values are retained.
> 
> However, when working with objects, any objects that contain the same data are considered to be different objects by Set because they have different object references — the actual “[pointers](https://developer.mozilla.org/en-US/docs/Mozilla/js-ctypes/Using_js-ctypes/Memory_Management)” in memory to those objects.

Reply:
Oh I see.. but for this particular use case (array of strings) it works perfectly fine and is probably the easiest way to do this

```js
const arr = ['abc', 'cbd', 'Abc', 'aBc', 'abc', 'cbd']
const uniqueArr = [...new Set(arr)]
console.log(uniqueArr) // ['abc', 'cbd', 'Abc', 'aBc']
```

Reply:
As yes, 4 messages later and I just noticed you're only storing the string. 🤦🏼  I'm marking this as resolved.